budgetportal/models.py

```diff
@@ -288,6 +297,29 @@ def get_govt_functions(self):
     def get_financial_year(self):
         return self.government.sphere.financial_year
 
+    def _get_latest_department_instance(self):
+        """ Try to find the department in the most recent year with the same slug.
+        Continue traversing backwards in time until found, or until the original year has been reached. """
+        years = FinancialYear.objects.all().order_by('-slug')
+        original_dept_year = self.government.sphere.financial_year
+        sphere_name = self.government.sphere.name
+        latest_existing_department = None
+        found_or_out_of_years, index = False, 0
+        while not found_or_out_of_years:
+            year = years[index]
+            index += 1
+            if year == original_dept_year:
+                found_or_out_of_years = True
+
+            latest_sphere = Sphere.objects.get(financial_year=year, name=sphere_name)
+            latest_government = latest_sphere.governments.get(name=self.government.name)
```

Comment:
if there's nothing else needing this intermediate functionality, wouldn't it be better to do something like
```python
Department.objects.filter(government__slug=self.government.slug,
                                        government__sphere__slug=self.government.sphere.slug,
...
)
```
?

Then we can avoid all this logic til it's needed

Reply:
Changed to your solution and had to add `slug=self.slug` to prevent other departments from being returned - thank you.